cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -1380,5 +1380,33 @@ class TestS3FSGeneric : public S3TestMixin, public GenericFileSystemTest {
 
 GENERIC_FS_TEST_FUNCTIONS(TestS3FSGeneric);
 
+////////////////////////////////////////////////////////////////////////////
+// S3GlobalOptionsDefaults tests
+
+class S3GlobalOptionsDefaults : public ::testing::Test {};
+
+TEST_F(S3GlobalOptionsDefaults, S3GlobalOptionsDefaultsLogLevel) {
+  // Verify we get the default value of Fatal
+  ASSERT_EQ(S3LogLevel::Fatal, arrow::fs::S3GlobalOptions::Defaults().log_level);
+
+  // Verify we get the value specified by env var and not the default
+  {
+    EnvVarGuard log_level_guard("ARROW_S3_LOG_LEVEL", "Error");
```

Comment:
How about using upper case style here because we use upper case in our document?

```suggestion
    EnvVarGuard log_level_guard("ARROW_S3_LOG_LEVEL", "ERROR");
```